messaging/src/main/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessor.java

```diff
@@ -672,13 +673,13 @@ public Builder tokenClaimInterval(long tokenClaimInterval) {
          * Defines the maximum number of segment this {@link StreamingEventProcessor} may claim. Defaults to {@value
          * Short#MAX_VALUE}.
          *
-         * @param maxClaimedSegments the maximum number fo claimed segments for this {@link StreamingEventProcessor}
+         * @param maxSegmentProvider the maximum number fo claimed segments for this {@link StreamingEventProcessor}
          *
          * @return the current Builder instance, for fluent interfacing
          */
-        public Builder maxClaimedSegments(int maxClaimedSegments) {
```

Comment:
For backwards compatibility we need to keep the maxClaimedSegments, but we can change the implementation to set the `maxSegmentProvider` instead.